aperturedb/DaskManager.py

```diff
@@ -17,6 +17,22 @@
 class DaskManager:
     def __init__(self, num_workers: int = -1):
         self.__num_workers = num_workers
+        # The -1 magic number is to use as many 90% of the cores (1 worker per core).
+        # This can be overridden by the user.
+        # Create a pool of workers.
+        # TODO: see if the same pool can be reused for multiple tasks.
+        workers = self.__num_workers if self.__num_workers != \
+            -1 else int(0.9 * mp.cpu_count())
+
+        self._cluster = LocalCluster(n_workers=workers)
+        self._cluster.shutdown_on_close = False
+        self._client = Client(self._cluster)
+        dask.config.set(scheduler="distributed")
```

Comment:
I see you switched the scope of methods for LocalCluster and Client, what was need for the change?

Reply:
dask distributed , although was doing all the ingestion stuff as the norm, however it was GCing the client and the cluster immediately after the context manager it earlier was confined to. I guess the len (which we use for assertions later) relies on this client, so we keep it in reference till the time were actually done.